examples/post-sync/main.go

```diff
@@ -0,0 +1,352 @@
+		}
+		syncID = SyncID{
+			Name:      root.Name,
+			Kind:      root.Kind,
+			Namespace: root.Namespace,
+		}
+		status = root.Status.Status
+		generation = root.Generation
+		observedGeneration = root.Status.ObservedGeneration
+		errs, commit = c.processRootSyncConditions(root.Status.Conditions)
+
+	case "RepoSync":
+		var repo v1beta1.RepoSync
+		if err := c.client.Get(ctx, req.NamespacedName, &repo); err != nil {
+			return reconcile.Result{}, client.IgnoreNotFound(err)
+		}
+		syncID = SyncID{
+			Name:      repo.Name,
+			Kind:      repo.Kind,
+			Namespace: req.Namespace,
+		}
+		status = repo.Status.Status
+		generation = repo.Generation
+		observedGeneration = repo.Status.ObservedGeneration
+		errs, commit = c.processRepoSyncConditions(repo.Status.Conditions)
+
+	default:
+		c.log.Error(fmt.Errorf("unknown sync kind: %s", c.syncKind), "unrecognized syncKind")
+		return reconcile.Result{}, fmt.Errorf("unknown sync kind: %s", c.syncKind)
+	}
+
+	// Process any errors from conditions
+	if err := c.handleErrors(syncID, errs, commit, generation, observedGeneration); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Process sync status
+	if !isSuccessfulSync(status) {
+		errors, commit := extractErrorAndCommitFromStatus(status)
+		if err := c.handleErrors(syncID, errors, commit, generation, observedGeneration); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// handleErrors aggregates error messages and logs them if they haven't been logged already.
+func (c *SyncStatusController) handleErrors(rsyncID SyncID, errMessage string, commit string, generation int64, observedGeneration int64) error {
+	if errMessage == "" {
+		return nil
+	}
+	if c.statusTracker.IsLogged(rsyncID, commit, errMessage) {
+		return nil
+	}
+	c.log.Info("Sync error detected", "sync", rsyncID.Name, "namespace", rsyncID.Namespace, "kind", rsyncID.Kind, "commit", commit, "status", "failed", "generation", generation, "observedGeneration", observedGeneration, "error", errMessage)
+
+	c.statusTracker.MarkLogged(rsyncID, commit, errMessage)
+	return nil
+}
+
+func (c *SyncStatusController) processRootSyncConditions(conditions []v1beta1.RootSyncCondition) (string, string) {
+	var errorMessages []string
+	var commit string
+
+	for _, condition := range conditions {
+		isErrorCondition := false
+		var conditionType v1beta1.RootSyncConditionType
+
+		switch condition.Type {
+		case v1beta1.RootSyncStalled:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+		case v1beta1.RootSyncReconcilerFinalizerFailure:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+			// Other condition types (Reconciling, Syncing, ReconcilerFinalizing)
+			// are not treated as primary error indicators here for aggregation.
+			// Stalled usually captures persistent source/rendering issues,
```

Comment:
I thought Stalled was actually just used by the reconciler-manager, not the reconciler. I thought source/rendering issues were just reported in the Syncing status...